libs/blocks/share/share.js

```diff
@@ -42,43 +42,62 @@ function getPlatforms(el) {
   });
 }
 
-function getDetails(name, url) {
-  switch (name) {
-    case 'facebook':
-      return { title: 'Facebook', href: `https://www.facebook.com/sharer/sharer.php?u=${url}` };
-    case 'twitter':
-      return { title: 'Twitter', href: `https://twitter.com/share?&url=${url}` };
-    case 'linkedin':
-      return { title: 'LinkedIn', href: `https://www.linkedin.com/sharing/share-offsite/?url=${url}` };
-    case 'pinterest':
-      return { title: 'Pinterest', href: `https://pinterest.com/pin/create/button/?url=${url}` };
-    default: return null;
-  }
-}
-
 export default async function decorate(el) {
   const config = getConfig();
   const base = config.miloLibs || config.codeRoot;
   const platforms = getPlatforms(el) || ['facebook', 'twitter', 'linkedin', 'pinterest'];
-  el.querySelector('div').remove();
-  const url = encodeURIComponent(window.location.href);
+  el.innerHTML = '';
+  const clipboardSupport = !!(navigator.clipboard)
+  if (clipboardSupport) platforms.push('clipboard');
   const svgs = await getSVGsfromFile(`${base}/blocks/share/share.svg`, platforms);
   if (!svgs) return;
 
-  const heading = await replaceKey('share-this-page', config);
   const toSentenceCase = (str) => (str && typeof str === 'string') ? str.toLowerCase().replace(/(^\s*\w|[\.\!\?]\s*\w)/g, (c) => c.toUpperCase()) : '';
-  el.append(createTag('p', null, toSentenceCase(heading)));
+  const shareToText = toSentenceCase(await replaceKey('share-to', config));
+  const url = encodeURIComponent(window.location.href);
+  const getDetails = (name, url) => {
+    switch (name) {
+      case 'facebook':
+        return { title: 'Facebook', href: `https://www.facebook.com/sharer/sharer.php?u=${url}` };
+      case 'twitter':
+        return { title: 'Twitter', href: `https://twitter.com/share?&url=${url}` };
+      case 'linkedin':
+        return { title: 'LinkedIn', href: `https://www.linkedin.com/sharing/share-offsite/?url=${url}` };
+      case 'pinterest':
+        return { title: 'Pinterest', href: `https://pinterest.com/pin/create/button/?url=${url}` };
+      default: return null;
+    }
+  };
+  
+  const heading = toSentenceCase(await replaceKey('share-this-page', config));
+  el.append(createTag('p', null, ((heading))));
   const container = createTag('p', { class: 'icon-container' });
-  svgs.forEach((svg) => {
-    const details = getDetails(svg.name, url);
-    if (!details) return;
-    const shareLink = createTag('a', { target: '_blank', href: details.href, title: `Share to ${details.title}` }, svg.svg);
+  svgs.forEach(async (svg) => {
```

Comment:
I assume there is only ever one `clipboard` svg, so it may be cleaner to pull it out of the svgs.

```
const clipboardSvg = svgs.find((svg) => svg.name === 'clipboard');
if (!(clipboardSupport && clipboardSvg)) return;

...
```